auto_nag/topcrash.py

```diff
@@ -212,24 +218,38 @@ def _format_criteria_names(criteria: List[dict]):
 class Topcrash:
     def __init__(
         self,
+        date: Optional[Union[str, datetime]] = "today",
+        duration: Optional[int] = 7,
         minimum_crashes: Optional[int] = 5,
         signature_block_patterns: list = CRASH_SIGNATURE_BLOCK_PATTERNS,
+        criteria: Iterable[dict] = TOP_CRASH_IDENTIFICATION_CRITERIA,
     ) -> None:
         """Constructor
 
         Args:
+            date: the final date. If not provided, the value will be today.
+            duration: the number of days to retrieve the crash data.
             minimum_crashes: the minimum number of crashes to consider a
                 signature in the top crashes.
             signature_block_list: a list of crash signature to be ignored.
+            criteria: the list of criteria to be used to query the top crashes.
         """
         self.minimum_crashes = minimum_crashes
         self.signature_block_patterns = signature_block_patterns
+        self.criteria = criteria
 
-    def _fetch_signatures_from_patters(self, patterns, date_range) -> Set[str]:
+        start_date = lmdutils.get_date(date, duration)
+        end_date = lmdutils.get_date(date)
+        self.date_range = socorro.SuperSearch.get_search_date(start_date, end_date)
+
+        self._blocked_signatures: Optional[Set[str]] = None
+
+    def _fetch_signatures_from_patters(self, patterns) -> Set[str]:
```

Comment:
```suggestion
    def _fetch_signatures_from_patterns(self, patterns) -> Set[str]:
```

Reply:
Nice catch. Since this typo has existed since before, I will fix it in a separate pull request. I filed a follow-up issue (#1649).